internal/service/quicksight/data_set.go

```diff
@@ -0,0 +1,2726 @@
+		UpdateContext: resourceDataSetUpdate,
+		DeleteContext: resourceDataSetDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"aws_account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidAccountID,
+			},
+
+			"column_groups": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 8,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"geo_spatial_column_group": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"columns": {
+										Type:     schema.TypeList,
+										Required: true,
+										MinItems: 1,
+										MaxItems: 16,
+										Elem: &schema.Schema{
+											Type:         schema.TypeString,
+											ValidateFunc: validation.StringLenBetween(1, 128),
+										},
+									},
+									"country_code": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(quicksight.GeoSpatialCountryCode_Values(), false),
+									},
+									"name": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringLenBetween(1, 64),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"column_level_permission_rules": {
+				Type:     schema.TypeList,
+				MinItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"column_names": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MinItems: 1,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"principals": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MinItems: 1,
+							MaxItems: 100,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+
+			"data_set_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"data_set_usage_configuration": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"disable_use_as_direct_query_source": {
+							Type:     schema.TypeBool,
+							Computed: true,
+							Optional: true,
+						},
+						"disable_use_as_imported_source": {
+							Type:     schema.TypeBool,
+							Computed: true,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"field_folders": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1000,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"field_folders_id": {
```

Comment:
```suggestion
						"field_folder_id": {
```
Similarly to the instance of using `physical_table_map_id` above, the ID here refers to the unique key for an individual field folder.